### PR TITLE
Added a new benefit area for .NET Core and some other smaller changes

### PR DESCRIPTION
**Pull request by davidcsa**

### CODE REVIEW

File: docs/core-concepts/dotnet-core.md [hunk 1]
```diff
@@ -23,16 +24,16 @@ Composition
...
 
 .NET APIs and Compatibility
 ---------------------------
 
-.NET Core can be thought of as a cross-platform version of the .NET Framework, at the layer of the .NET Framework Base Class Libraries (BCL). It implements the [.NET Standard Library](../concepts/dotnet-standard-library.md) specification. .NET Core provides a subset of the APIs that are available in the .NET Framework or Mono/Xamarin. In some cases, types are not fully implemented (some members are not available or have been moved).
+.NET Core can be thought of as a cross-platform version of the .NET Framework, at the layer of the .NET Framework Base Class Libraries (BCL). It implements the [.NET Standard Library](../concepts/dotnet-standard-library.md) specification. .NET Core provides a subset of the APIs that are available in the .NET Framework or Mono/Xamarin.
 
-Look at the [.NET Core roadmap](https://github.com/dotnet/core/blob/master/roadmap.md) to learn more about the plan to expand the .NET Core BCL API.
+Look at the [.NET Core roadmap](https://github.com/dotnet/core/blob/master/roadmap.md) to learn more about the APIs supported by .NET Core and the roadmap to increase to additional APIs..
```
Comment (richlander) [inline docs/core-concepts/dotnet-core.md]: The roadmap doesn't include any info about existing APIs. It is only forward-looking.

File: docs/core-concepts/dotnet-core.md [hunk 2]
```diff
@@ -48,7 +49,6 @@ By itself, .NET Core includes a single application model: console apps, which ar
 
 - [ASP.NET Core](http://asp.net)
 - [Windows 10 Universal Windows Platform (UWP)](https://developer.microsoft.com/windows)
-- [Xamarin.Forms](https://www.xamarin.com/forms)
```
Comment (richlander) [inline docs/core-concepts/dotnet-core.md]: Why remove this one?

**Comment by richlander** [comment]
LGTM modulo comment

**Comment by mairaw** [comment]
@davidcsa are you planning to modify this so we can merge?

**Comment by richlander** [comment]
These files have moved. I made a similar change elsewhere.

**Pull request closed without merging**